File: packages/core/src/infra/tokenizer/logic/tokenizer-asset-paths.ts
/**
 * Tokenizer asset path table — platform-agnostic relative paths per family.
 *
 * @module infra/tokenizer/logic/tokenizer-asset-paths
 */

import type { TokenizerFamily } from "../ports/token-counter.port.js";

/** Relative asset paths per tokenizer family (ST parity). */
export function tokenizerAssetPaths(family: TokenizerFamily): {
  readonly primary: string;
  readonly fallback?: string;
  readonly kind: "json" | "model";
} | null {
  switch (family) {
    case "claude":
      return { primary: "claude.json", kind: "json" };
    case "llama3":
      return { primary: "llama3.json", kind: "json" };
    case "llama":
      return { primary: "llama.model", kind: "model" };
    case "mistral":
      return { primary: "mistral.model", kind: "model" };
    case "yi":
      return { primary: "yi.model", kind: "model" };
    case "gemma":
      return { primary: "gemma.model", kind: "model" };
    case "jamba":
      return { primary: "jamba.model", kind: "model" };
    case "qwen2":
      return { primary: "web/qwen2.json", fallback: "llama3.json", kind: "json" };
    case "command-r":
      return { primary: "web/command-r.json", fallback: "llama3.json", kind: "json" };
    case "command-a":
      return { primary: "web/command-a.json", fallback: "llama3.json", kind: "json" };
    case "nemo":
      return { primary: "web/nemo.json", fallback: "llama3.json", kind: "json" };
    case "deepseek":
      return { primary: "web/deepseek.json", fallback: "llama3.json", kind: "json" };
    default:
      return null;
  }
}

File: packages/tokenizer-driver-rn/android/src/main/java/com/novelmaster/tokenizer/TokenizerAssetPaths.kt
package com.novelmaster.tokenizer

/** Asset paths under `assets/tokenizers/` — mirrors core `tokenizerAssetPaths`. */
internal data class AssetPathSpec(
  val primary: String,
  val fallback: String? = null,
  val kind: String,
)

internal object TokenizerAssetPaths {
  fun forFamily(family: String): AssetPathSpec? {
    return when (family) {
      "claude" -> AssetPathSpec("claude.json", kind = "json")
      "llama3" -> AssetPathSpec("llama3.json", kind = "json")
      "llama" -> AssetPathSpec("llama.model", kind = "model")
      "mistral" -> AssetPathSpec("mistral.model", kind = "model")
      "yi" -> AssetPathSpec("yi.model", kind = "model")
      "gemma" -> AssetPathSpec("gemma.model", kind = "model")
      "jamba" -> AssetPathSpec("jamba.model", kind = "model")
      "qwen2" -> AssetPathSpec("web/qwen2.json", "llama3.json", "json")
      "command-r" -> AssetPathSpec("web/command-r.json", "llama3.json", "json")
      "command-a" -> AssetPathSpec("web/command-a.json", "llama3.json", "json")
      "nemo" -> AssetPathSpec("web/nemo.json", "llama3.json", "json")
      "deepseek" -> AssetPathSpec("web/deepseek.json", "llama3.json", "json")
      else -> null
    }
  }
}
